inst/shiny/modules/column_mapping.R

```diff
@@ -1,3 +1,42 @@
+#' Column Mapping Widget
+#'
+#' A reusable UI component for mapping dataset columns to specific identifiers or roles.
+#'
+#' @param ns A namespace function to generate IDs for Shiny inputs.
+#' @param id A string representing the unique identifier for the widget.
+#' @param tooltip_text A string containing the tooltip text to guide users.
+#'
+#' @return A Shiny `div` containing a `selectizeInput` with associated labels and tooltip.
+#'
+#' @examples
+#' column_mapping_widget(ns = NS("example"), id = "STUDYID",
+#' tooltip_text = "Select the study identifier column.")
+#'
+#' @export
+column_mapping_widget <- function(ns, id, tooltip_text) {
```

Comment:
suggestion: I would suggest prefacing this function name with a dot `.`, to signify that is is a _local_ function and used only within the scope of this module.
```suggestion
.column_mapping_widget <- function(ns, id, tooltip_text) {
```